File: src/Striver_SDE_Sheet/Stack_and_Queue/SortStack/Sort_Stack.java
package Striver_SDE_Sheet.Stack_and_Queue.SortStack;
import java.util.*;

public class Solution {

	static void sortedInsert(Stack<Integer> stack , int num){
		if(stack.empty() || (!stack.empty() && stack.peek()<num)){
			stack.push(num) ;
			return ;
		}
		int n = stack.peek();
		stack.pop();
		
		sortedInsert(stack , num);
		stack.push(n);
	}
	
	public static void sortStack(Stack<Integer> stack) {
		if(stack.empty()){
			return ;
		}
		int num = stack.peek();
		stack.pop();
		sortStack(stack);
		sortedInsert(stack , num);
	}
	
}

File: src/Striver_SDE_Sheet/Stack_and_Queue/SortStack/Sort_Stack.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    void insertInSorted(stack<int> &stack,int element){
        if(stack.empty() || (!stack.empty() && stack.top()<element)){
            stack.push(element);
            return;
        }
        int num = stack.top();
        stack.pop();
        
        insertInSorted(stack,element);
        stack.push(num);
    }
    void sortStack(stack<int> &stack) {
        if(stack.empty()){
            return;
        }
        int num = stack.top();
        stack.pop();
        sortStack(stack);
        insertInSorted(stack,num);
    }
};